### PR TITLE
Update gophertunnel to 1.16.210

### PR DESCRIPTION
**Pull request by TwistedAsylumMC**

### CODE REVIEW

**Comment by JustTalDevelops** [comment]
Seems like you're missing the PlayFabID in start game / game data

File: minecraft/protocol/packet/clientbound_debug_renderer.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	// Position is the position to spawn the debug on.
+	Position mgl32.Vec3
+	// Red is the red value from the RGBA colour rendered on the debug.
+	Red float32
+	// Green is the green value from the RGBA colour rendered on the debug.
+	Green float32
+	// Blue is the blue value from the RGBA colour rendered on the debug.
+	Blue float32
+	// Alpha is the alpha value from the RGBA colour rendered on the debug.
+	Alpha float32
```
Comment (Sandertv) [inline minecraft/protocol/packet/clientbound_debug_renderer.go:28]: Are these 0-1? Might be good to document that.

File: minecraft/protocol/packet/clientbound_debug_renderer.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+func (*ClientBoundDebugRenderer) ID() uint32 {
+	return IDClientBoundDebugRenderer
+}
+
+// Marshal ...
+func (pk *ClientBoundDebugRenderer) Marshal(w *protocol.Writer) {
+	w.Int32(&pk.Type)
+
+	switch pk.Type {
+	case ClientBoundDebugRendererTypeClear:
```
Comment (Sandertv) [inline minecraft/protocol/packet/clientbound_debug_renderer.go:43]: Is this case supposed to do nothing? In that case you can just remove it.

File: minecraft/protocol/packet/clientbound_debug_renderer.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		w.Int64(&pk.Duration)
+	}
+}
+
+// Unmarshal ...
+func (pk *ClientBoundDebugRenderer) Unmarshal(r *protocol.Reader) {
+	r.Int32(&pk.Type)
+
+	switch pk.Type {
+	case ClientBoundDebugRendererTypeClear:
```
Comment (Sandertv) [inline minecraft/protocol/packet/clientbound_debug_renderer.go:60]: Same as above.

File: minecraft/protocol/packet/player_auth_input.go [hunk 4]
```diff
@@ -107,6 +125,43 @@ func (pk *PlayerAuthInput) Marshal(w *protocol.Writer) {
...
+			protocol.InvAction(w, &a, pk.ItemInteractionData.HasNetworkIDs)
+		}
+		w.Varuint32(&pk.ItemInteractionData.ActionType)
+		w.BlockPos(&pk.ItemInteractionData.BlockPosition)
+		w.Varint32(&pk.ItemInteractionData.BlockFace)
+		w.Varint32(&pk.ItemInteractionData.HotBarSlot)
+		w.Item(&pk.ItemInteractionData.HeldItem)
+		w.Vec3(&pk.ItemInteractionData.Position)
+		w.Vec3(&pk.ItemInteractionData.ClickedPosition)
+		w.Varuint32(&pk.ItemInteractionData.BlockRuntimeID)
```
Comment (Sandertv) [inline minecraft/protocol/packet/player_auth_input.go:151]: Would be useful to pull all this out in its own function.

File: minecraft/protocol/packet/player_auth_input.go [hunk 5]
```diff
@@ -124,4 +179,43 @@ func (pk *PlayerAuthInput) Unmarshal(r *protocol.Reader) {
...
+			protocol.InvAction(r, &pk.ItemInteractionData.Actions[i], pk.ItemInteractionData.HasNetworkIDs)
+		}
+		r.Varuint32(&pk.ItemInteractionData.ActionType)
+		r.BlockPos(&pk.ItemInteractionData.BlockPosition)
+		r.Varint32(&pk.ItemInteractionData.BlockFace)
+		r.Varint32(&pk.ItemInteractionData.HotBarSlot)
+		r.Item(&pk.ItemInteractionData.HeldItem)
+		r.Vec3(&pk.ItemInteractionData.Position)
+		r.Vec3(&pk.ItemInteractionData.ClickedPosition)
+		r.Varuint32(&pk.ItemInteractionData.BlockRuntimeID)
```
Comment (Sandertv) [inline minecraft/protocol/packet/player_auth_input.go:206]: Some as above for all this.

**Review by Sandertv — CHANGES_REQUESTED** [review]
Got a couple of comments, also don't forget to add the PlayFabId field in the login.ClientData struct.

File: minecraft/protocol/packet/clientbound_debug_renderer.go (hunk 1 above)
Comment (Mohamed587100) [reply minecraft/protocol/packet/clientbound_debug_renderer.go:28]: literally it tell us this is ARGB but each channel is store as uint8([reference](https://en.m.wikipedia.org/wiki/RGBA_color_model))

Comment (Sandertv) [reply minecraft/protocol/packet/clientbound_debug_renderer.go:28]: Except, as the fields indicate, these are float32s, not uint8s.

File: minecraft/protocol/packet/clientbound_debug_renderer.go (hunk 2 above)
Comment (Mohamed587100) [reply minecraft/protocol/packet/clientbound_debug_renderer.go:43]: change to 
`
if pk.Type == ClientBoundDebugRendererTypeAddCube {
		w.String(&pk.Text)
		w.Vec3(&pk.Position)
		w.Float32(&pk.Red)
		w.Float32(&pk.Green)
		w.Float32(&pk.Blue)
		w.Float32(&pk.Alpha)
		w.Int64(&pk.Duration)
	}
`

File: minecraft/protocol/packet/clientbound_debug_renderer.go (hunk 3 above)
Comment (Mohamed587100) [reply minecraft/protocol/packet/clientbound_debug_renderer.go:60]: also
```go
if pk.Type == ClientBoundDebugRendererTypeAddCube {
		r.String(&pk.Text)
		r.Vec3(&pk.Position)
		r.Float32(&pk.Red)
		r.Float32(&pk.Green)
		r.Float32(&pk.Blue)
		r.Float32(&pk.Alpha)
		r.Int64(&pk.Duration)
}
```

**Review by Sandertv — APPROVED** [review]

**Pull request merged**